packages/studio-plugin/src/createStudioStylingPlugin.ts

```diff
@@ -0,0 +1,49 @@
+import { PluginOption } from "vite";
+
+/**
+ * Adds the ?inline query parameter to all CSS imports
+ * unless they are CSS from Studio, which is determined
+ * from the path of its importer. This prevents user styling
+ * from affecting Studio UI.
+ */
+export default function createStudioStylingPlugin(): PluginOption {
+  return {
+    name: "yext-studio-style-plugin",
+    enforce: "pre",
+    async resolveId(id, importer) {
+      if (!importer) {
+        return;
+      }
+      if (isImportedByStudio(importer)) {
+        return;
+      }
+      if (isStyleFile(id)) {
+        const resolvedId = (
+          await this.resolve(id, importer, { skipSelf: true })
+        )?.id;
+        if (resolvedId) {
+          return addQueryParameter(resolvedId, "inline");
+        }
+        throw new Error(
+          "Styling file could not be resolved by yext-studio-style-plugin"
+        );
+      }
+    },
+  };
+}
+
+function isImportedByStudio(id: string) {
+  return !!(id.match(/.*@yext\/studio.*/) || id.match(/.*packages\/studio*/));
```

Comment:
what are some example paths here? for the first regex, would that always be a path in node_modules? and for the second, is there some way to make sure the path is in our monorepo (e.g. check relative to the current file's path or something)?

Reply:
an example would be something like:
`/Users/ataing/Documents/yext-marketing-creative/node_modules/@yext/studio/index.html`
It will always be in node modules!

For our monorepo check, I made it so that we derive the packages dir from our current path, and ignore all the css imports that originate from under that directory!

Reply:
can we update the first check to make sure the path is within `node_modules`?

Reply:
done!